src/test/java/org/broadinstitute/hellbender/engine/filters/OverclippedReadFilterUnitTest.java

```diff
@@ -24,22 +24,131 @@
+        final Cigar cigar = TextCigarCodec.decode(cigarString);
+        return ArtificialReadUtils.createArtificialRead(header, cigar);
+    }
 
     @Test(dataProvider= "OverclippedDataProvider")
-    public void testOverclippedFilter(final String cigarString, boolean doNotRequireSoftClipsOnBothEnds, final boolean expectedResult) {
+    public void testOverclippedFilter(final String cigarString,
+                                      final boolean doNotRequireSoftClipsOnBothEnds,
+                                      final boolean expectedResult) {
 
-        final OverclippedReadFilter filter = new OverclippedReadFilter(30, false);
-        filter.doNotRequireSoftClipsOnBothEnds = doNotRequireSoftClipsOnBothEnds;
+        final OverclippedReadFilter filter = new OverclippedReadFilter(30, doNotRequireSoftClipsOnBothEnds);
         final GATKRead read = buildSAMRead(cigarString);
         Assert.assertEquals(filter.test(read), expectedResult, cigarString);
     }
 
-    private GATKRead buildSAMRead(final String cigarString) {
-        final Cigar cigar = TextCigarCodec.decode(cigarString);
-        return ArtificialReadUtils.createArtificialRead(header, cigar);
+    @Test(dataProvider= "OverclippedSoftClipRatioDataProvider")
+    public void testOverclippedSoftClipRatioFilter(final String cigarString,
+                                                   final double clipRatio,
+                                                   final boolean expectedResult) {
+
+        final OverclippedReadFilter filter = new OverclippedReadFilter(10);
+        filter.minimumSoftClippedRatio = clipRatio;
+
+        final GATKRead read = buildSAMRead(cigarString);
+        Assert.assertEquals(filter.test(read), expectedResult, cigarString);
+    }
+
+    @Test(dataProvider= "OverclippedLeadingTrailingSoftClipRatioDataProvider")
+    public void testOverclippedLeadingTrailingSoftClipRatioFilter(final String cigarString,
+                                                   final double clipRatio,
+                                                   final boolean expectedResult) {
+
+        final OverclippedReadFilter filter = new OverclippedReadFilter(10);
+        filter.minimumLeadingTrailingSoftClippedRatio = clipRatio;
+
+        final GATKRead read = buildSAMRead(cigarString);
+        Assert.assertEquals(filter.test(read), expectedResult, cigarString);
+    }
+
+    @DataProvider(name = "OverclippedSoftClipRatioDataProvider")
+    public Iterator<Object[]> overclippedSoftClipRatioDataProvider() {
+        final List<Object[]> testData = new LinkedList<>();
+
+        // ---------------------------------------
+        // Null / trivial cases:
+        testData.add(new Object[] { "", 0.1, false });
+        testData.add(new Object[] { "10H", 0.1, false });
+
+        // ---------------------------------------
+        // Min bases test:
+        testData.add(new Object[] { "20S5M", 0.2, false }); // 20/25 = .8
```

Comment:
They were showing that strings below 30 bases will return false even though by the other metric they would pass.  The other tests have a smaller minAlignedBases and all pass because the threshold has been hit.

These tests have since been removed anyway because of the new class structure.